sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/codegen/UTF8StringBuilder.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.expressions.codegen;
+
+import java.nio.charset.StandardCharsets;
+
+import org.apache.spark.unsafe.types.UTF8String;
+
+/**
+ * A helper class to write `UTF8String`, `String`, and `byte[]` data into an internal buffer
+ * and get a final concatenated string.
+ */
+public class UTF8StringBuilder {
+
+  private StringBuilder buffer;
```

Comment:
@cloud-fan In the latest commit, I just made this class by just wrapping `StringBuider`. Or, we can use byte[]-backed buffer implemenation like https://github.com/apache/spark/pull/20024/commits/91df078a99b6ec5f2063b2e73170336e3fe812d1#diff-ef9a1d6680a577c946ac779019a13beaR28.
Which one is preferred?

Reply:
We should use the byte[] backed implementation. At the end we need a `UTF8String`, by wrapping `StringBuilder` we still have the unnecessary `String` -> `UTF8String` -> `String` conversion.

Reply:
ok